### ISSUE TITLE
Publish to microsoft/winget-pkgs

### ISSUE THREAD

**Issue by Mannshoch** [IfcOpenShell/IfcOpenShell#2362]
Please publish to https://github.com/microsoft/winget-pkgs

so we could install from Windows Commandline e.g.
```cmd
winget install ifcOpenShell.BlenderBIM
winget install ifcOpenShell.IfcConvert
winget install IfcOpenShell.IfcOpenShell-python
```

It should download install and add them to PATH

**Comment by aothms** [comment]
There is some work on chocolatey for blenderbim specifically. I don't think we have the resources to investigate all these platforms, but contributions are welcome. https://github.com/IfcOpenShell/IfcOpenShell/pull/2335

**Comment by Moult** [comment]
Here's the link to the (pending manual moderation) package for Chocolatey courtesy of @fbpyr :) https://community.chocolatey.org/packages/blenderbim-nightly/

I guess with a bit of tweaks to the work already done by @fbpyr it can ship IfcConvert/IfcOpenShell-Python too. Would you be interested in contributing?

**Comment by aothms** [comment]
The daily built conda packages from @Krande  also include IfcConvert (and ofc IfcOpenShell-Python), so if you do `conda install -c ifcopenshell ifcopenshell` you also have a working IfcConvert added to your path as well as IfcOpenShell-Python.

**Comment by Krande** [comment]
@aothms and others trying to install the daily conda builds of IfcOpenShell. Just a heads up.

You will have to add the `conda-forge` to your channel source in addition to the `ifcopenshell` channel:

`conda install -c ifcopenshell -c conda-forge ifcopenshell` 

Otherwise you will get `UnsatisfiableError` which is basically saying that it can't find all the necessary conda package dependencies of IfcOpenShell in the IfcOpenShell conda channel.

**Comment by fbpyr** [comment]
@Moult  Would be nice indeed, to have IfcConvert available from choco as well. 🙂 
@Krande where does the IfcConvert.exe [daily build](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/.github/workflows/ci-ifcopenshell-conda-daily.yml) end up? 
is it directly downloadable from somewhere on conda? 
(or should we use the binaries published on [ifcopenshell.org](http://www.ifcopenshell.org/ifcconvert) ? )
If so, then the mechanism could be similar to choco blenderbim-nightly. 🤔

**Comment by Krande** [comment]
@fbpyr It should be accessible in the conda environment in which you install ifcopenshell.

So if you do like this (I just tested this and it worked on my computer):

`conda create -n testenv -c ifcopenshell -c conda-forge ifcopenshell`

It will create a conda environment called `testenv` and install the latest IfcOpenShell daily build.

After the installation is complete you can activate the environment with 
`activate testenv` and check if IfcConvert is accessible from path:

![image](https://user-images.githubusercontent.com/9642232/188387454-b2dcf0d7-e5fd-451f-9064-42c6bf55f662.png)

**Comment by fbpyr** [comment]
@Krande thank you for quick reply & explanation. 🙂  
Since the gh-action choco blenderbim nightly is currently a ubuntu container, conda would correctly install a linux binary, 
which would not be very useful for choco. So I guess options would be: 
* run a separate choco ifcconvert gh action on a win container image and use that ifcconvert.exe binary. 
* use existing blenderbim or new gh action on linux container and just pull the binary from direct download link from http://www.ifcopenshell.org/ifcconvert 

@Krande is ifcconvert on http://www.ifcopenshell.org/ifcconvert updated by the [conda daily build](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/.github/workflows/ci-ifcopenshell-conda-daily.yml) as well?

**Comment by Krande** [comment]
@fbpyr I believe the ifcconvert executables in http://www.ifcopenshell.org/ifcconvert are built by [IfcOpenBot](https://github.com/IfcOpenShell/IfcOpenBot). I am not sure how often those executables are updated, but my guess would be not as often as the daily builds.

**Comment by fbpyr** [comment]
@Krande ah I see - thank you for link and explanation! 🙂

**Comment by aothms** [comment]
Also,  the ifcopenbot builds are statically linked, so they are self contained (fat) executable.

The conda builds depend on other conda libs provided as dynamic linked libraries, so need to be copied along. You could try something like `copypedeps` (https://github.com/brechtsanders/pedeps) for that.

**Comment by fbpyr** [comment]
Thank you for pointing that out, @aothms !  
Packaging it for choco should then ideally use the statically linked IfcConvert.exe, I guess. 
(not necessarily a must, but would be easier to handle)
(pedeps sounds interesting. would be even nicer if it could also be re-packaged with it, 
into one exe including the dependencies. 🤔  )
So maybe it would be a good starting point to use the ones generated by ifcopenbot. (?)

**Comment by aothms** [comment]
>  So maybe it would be a good starting point to use the ones generated by ifcopenbot. (?)

I agree, that's also most consistent with the approach in blenderbim. And it's a bit more feature complete because the conda build lacks Collada (.dae) support atm.

**Comment by Moult** [comment]
I have been abusing Makefiles to provide aliases to a set of bash commands, so `make dist` makes a distributable. For example [this line](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/Makefile#L70) fetches the appropriate latest bot build. Every so often, typically after I battle test a bot build a bit, I bump that commit sha, so it lags a little behind the bot build, but it is hand-vetted. Feel free to reuse that oneliner to download the latest IfcConvert.exe and package it for chocolatey, and then Github actions can just run "make dist-choco" or something, so I'll remember to bump it.

Maybe a better long term solution is to somehow know what the latest successful (across all platforms, since sometimes one or two fail) build is, but I don't know Github actions well enough to know how to do that.

**Comment by fbpyr** [comment]
@Moult neat! Thank you - I will certainly use something like that in the python script, 
which bootstraps the choco spec with link and sha256sum.. 🙂 
As the (current) latest link on on http://www.ifcopenshell.org/ifcconvert points directly to: 
https://s3.amazonaws.com/ifcopenshell-builds/IfcConvert-v0.7.0-e508fb4-win64.zip 
I am thinking of using the provided git-sha as well (`v0.7.0-e508fb4`) in the 
chocolatey-version to submit. 
That way we can compare and know we only would need to upload to choco, 
if the git-sha actually changed. 🤔

**Comment by Moult** [comment]
@fbpyr yeah I'm not quite sure how the ifcopenshell.org site works but that's not the latest version, the latest should be this commit: https://github.com/IfcOpenBot/IfcOpenShell/commit/dc67287d4b083cabae39ee4f92fa93b6f74cae11

Including the sha is a good idea!
